pkgs/build-support/emacs/wrapper.nix

```diff
@@ -193,33 +193,40 @@ runCommand
     # Wrap emacs and friends so they find our site-start.el before the original.
     for prog in $emacs/bin/*; do # */
       local progname=$(basename "$prog")
-      rm -f "$out/bin/$progname"
+      rm -f "$out/bin/$progname" "$out/bin/.$progname-wrapped"
 
-      substitute ${./wrapper.sh} $out/bin/$progname \
+      substitute ${./wrapper.sh} $out/bin/.$progname-wrapped \
         --subst-var-by bash ${emacs.stdenv.shell} \
         --subst-var-by wrapperSiteLisp "$deps/share/emacs/site-lisp" \
         --subst-var-by wrapperSiteLispNative "$deps/share/emacs/native-lisp" \
         --subst-var prog
-      chmod +x $out/bin/$progname
+      chmod +x $out/bin/.$progname-wrapped
+      # Create a “NOP” binary wrapper for the pure sake of it becoming a
+      # non-shebang, actual binary. See the makeBinaryWrapper docs for rationale
+      # (summary: it allows you to use emacs as a shebang itself on Darwin,
+      # e.g. #!$ {emacs}/bin/emacs --script)
+      makeBinaryWrapper $out/bin/.$progname-wrapped $out/bin/$progname
```

Comment:
Why not use `wrapProgram` instead of dealing with names directly?

Reply:
Because the -wrapped binary is created by a separate bespoke wrapper script, I think.